codegen/table.go

```diff
@@ -4,13 +4,17 @@ import (
 	"reflect"
 
 	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/grpc-ecosystem/go-grpc-middleware/providers/zerolog/v2"
 )
 
 type ResourceDefinition struct {
 	Name  string
 	Table *TableDefinition
 }
 
+type NameTransformer func(reflect.StructField) (string, error)
+type TypeTransformer func(reflect.Type) (schema.ValueType, error)
```

Comment:
I think both @hermanschaaf and @candiduslynx suggested to extend this API to allow either the field or any type.

Reply:
if it will be `any` then the codegen will need to potentially invoke the type and create the objects instead of the custom transformer to do `reflect.TypeOf`. I need to see the code though for the `any` but I feel it might be more prone to issues. `StructField` is fine imo I was on the fence here but I can go with StructField so the transformer will have all the context if necessary. 

cc @hermanschaaf @candiduslynx

Reply:
Actually second though I removed the implicit override behaviour and made `DefaultTypeTransformer` public so the user can decide for example on the order and so on. this also make the sdk code cleaner and more explicit.

Reply:
using `any` enables easier and more confident implementation by the developers, as they can just do a type-check (or type switch) and handle accordingly. Otherwise they'll need to learn about the `reflect` package.

Reply:
I would still vote for `StructField`, in order to provide all the necessary context, even if it means having to learn about the `reflect` package. Otherwise, we're bound to run into cases where we need that context but don't have it. (I'm thinking especially about cases where the name of the field indicates something, like an IP or a date, so you want to override it, but the type of the field is string)

Reply:
Im in with @hermanschaaf here. I think developers that uses codegen will anyhow need to learn about reflect either way. Will update and request re-review.

Reply:
I finally agree that `refrect.StructField` allows for more interesting use-cases, so let's stick with it